interfaces/src/txpool.rs

```diff
@@ -0,0 +1,127 @@
+
+    /// find all dependent tx and return them with requsted dependencies in one list sorted by Price.
+    async fn find_dependent(&self, hashes: &[TxId]) -> Vec<Arc<Transaction>>;
+
+    /// Iterete over `hashes` and return all hashes that we dont have.
+    /// Needed when we receive list of new hashed from peer with
+    /// **BroadcastTransactionHashes**, so txpool needs to return
+    /// tx that we dont have, and request them from that particular peer.
+    async fn filter_by_negative(&self, hashes: &[TxId]) -> Vec<TxId>;
+
+    /// Return all sorted transactions that are includable in next block.
+    /// This is going to be heavy operation, use it with only when needed.
+    async fn includable(&self) -> Vec<Arc<Transaction>>;
+
+    /// When block is updated we need to receive all spend outputs and remove them from txpool
+    /// There is three posibilities here. New Block is added or block is reverted/slashed from
+    ///  chain or both. When new block is added we need to remove transactions from pool.
+    /// Un case of block reverting we need to reinsert transaction in pool. What usually happens
+    /// with reorg is that one block get reverted and one new gets add, in that sense it is highly
+    /// probably that a lof of transactions that got reverted are going to be reinserted in new block.
+    /// That explains why we have only one function and it contains two list of reverted/inserted
+    /// transactions
+    async fn block_update(&self /*spend_outputs: [Input], added_outputs: [AddedOutputs]*/);
+
+    /// remove transaction from pool needed on user demand. Low priority
+    async fn remove(&self, hashes: &[TxId]);
+
+    async fn subscribe(&self, sub: Arc<dyn Subscriber>);
+}
+
+#[derive(Error, Debug, PartialEq, Eq, Clone)]
+pub enum Error {
+    #[error("Transaction is not inserted. Hash is already known")]
+    NotInsertedTxKnown,
+    #[error("Transaction is not inserted. Pool limit is hit, try to increase gas_price")]
+    NotInsertedLimitHit,
+    #[error("TxPool required that transaction contains metadata")]
+    NoMetadata,
+    #[error("Transaction is not inserted. More priced tx {0:?} already spend this UTXO output: {1:?}. Price should be bumped by 12.5%")]
+    NotInsertedCollision(TxId, UtxoId),
+    #[error("Transaction is not inserted. Dependent UTXO output is not existing: {0:?}")]
+    NotInsertedOutputNotExisting(UtxoId),
+    #[error("Transaction is not inserted. UTXO input contract is not existing: {0:?}")]
+    NotInsertedInputContractNotExisting(ContractId),
+    #[error("Transaction is not inserted. UTXO is not existing: {0:?}")]
+    NotInsertedInputTxNotExisting(TxId),
+    #[error(
+        "Transaction is not inserted. UTXO requires Contract input {0:?} that is priced lower"
+    )]
+    NotInsertedContractPricedLower(ContractId),
+    #[error("Transaction is not inserted. Input output missamatch. Coin owner is different from expected input")]
+    NotInsertedIOWrongOwner,
```

Comment:
Capitalizing `IO` isn't following UpperCamelCase:

> In UpperCamelCase, acronyms and contractions of compound words count as one word: use Uuid rather than UUID, Usize rather than USize or Stdin rather than StdIn

https://rust-lang.github.io/api-guidelines/naming.html

Reply:
sure